### PR TITLE
Refactor(eos_designs): Add helper to retrieve ip from ip prefix

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

Address TODO in the WAN code for 5.0.0

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

one helper function in shared utils to consistantly do the operation rather than several slightly different ways over the place (maxsplit / not maxsplit)

## How to test

no changes in molecule

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/_eos_designs/shared_utils/utils.py [hunk 1]
```diff
@@ -72,3 +72,7 @@ def get_merged_adapter_settings(self: SharedUtils, adapter_or_network_port_setti
         profile_name = adapter_or_network_port_settings.get("profile")
         adapter_profile = self.get_merged_port_profile(profile_name)
         return merge(adapter_profile, adapter_or_network_port_settings, list_merge="replace", destructive_merge=False)
+
+    def get_ip_from_ip_prefix(self: SharedUtils, ip_prefix: str) -> str:
+        """Return the ip part of an ip/mask prefix."""
+        return ip_prefix.split("/", maxsplit=1)[0]
```
Comment (MaheshGSLAB) [inline python-avd/pyavd/_eos_designs/shared_utils/utils.py]: Can we use the python module to get the ip address from ip prefix
```
from ipaddress import ip_interface
return (str(ip_interface(ip_prefix).ip))
```

Comment (gmuloc) [reply python-avd/pyavd/_eos_designs/shared_utils/utils.py]: could do - I guess it all depends which one would be faster

Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_designs/shared_utils/utils.py]: yes current one is more faster then ip_interface

**Comment by laxmikantchintakindi** [comment]
Please use this method on line 720 in python-avd/pyavd/_eos_designs/structured_config/network_services/utils.py

**Comment by laxmikantchintakindi** [comment]
Also please check if we can replace the method `ip_interface` wherever it is used.
e.g. structured_config/overlay/management_cvx.py

File: python-avd/pyavd/_eos_designs/shared_utils/utils.py [hunk 2]
```diff
@@ -72,3 +72,7 @@ def get_merged_adapter_settings(self: SharedUtils, adapter_or_network_port_setti
         profile_name = adapter_or_network_port_settings.get("profile")
         adapter_profile = self.get_merged_port_profile(profile_name)
         return merge(adapter_profile, adapter_or_network_port_settings, list_merge="replace", destructive_merge=False)
+
+    def get_ip_from_ip_prefix(self: SharedUtils, ip_prefix: str) -> str:
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_designs/shared_utils/utils.py]: This should be moved to _utils and imported instead of having it in shared_utils.

**Review by ClausHolbechArista — DISMISSED** [review]

File: python-avd/pyavd/_eos_designs/shared_utils/utils.py (hunk 2 above)
Comment (ClausHolbechArista) [reply python-avd/pyavd/_eos_designs/shared_utils/utils.py]: done

**Review by ClausHolbechArista — APPROVED** [review]

**Review by laxmikantchintakindi — APPROVED** [review]

**Pull request merged**